src/linter/LinterContext.ts

```diff
@@ -368,6 +368,15 @@ export default class LinterContext {
 			lintResults.push(this.generateLintResult(resourcePath));
 		}
 
+		// Provide deterministic order of results & messages.
+		// Some of the (virtual) files might be generated during linting
+		// (XML transpilation & extraction) and their order might be non-deterministic.
+		lintResults.sort((a, b) => a.filePath.localeCompare(b.filePath));
```

Comment:
Sorting by filePath is already done in `src/linter/linter.ts` after the original filePath has been added. I'd prefer to keep it that way, as it might have differences based on the real filesystem paths (e.g. `src` / `test`, `webapp`, ...)